### ISSUE TITLE
Assert failure when using generic wxDataViewListCtrl with drag and drop on Linux

### ISSUE THREAD

**Issue by taler21** [wxWidgets/wxWidgets#22213]
This issue can also be seen in the **dataview sample** when using the **generic `wxDataViewListCtrl`** on Linux.
Note that by default the native DVC is used in wxGTK.

![af_enable-drag-source](https://user-images.githubusercontent.com/99262969/159273991-cde8391c-4ac2-4b29-b6c4-66ff2e067a50.png)

![af_dragging](https://user-images.githubusercontent.com/99262969/159274027-11e09ff6-ea90-4782-a573-913754375a70.png)

With wxWidgets-3.1.5 it works fine. So it's a regression in the master.

**Platform and version information**
 - wxWidgets version you use: **Latest master** for upcoming 3.1.6
 - wxWidgets port you use: wxGTK (GTK2)

**Comment by taler21** [comment]
You only have to start the **dataview sample**.
Or drag an item from the list in the MyMusicTreeModel page. Both result in an assert failure.

But you have to use the **generic** `wxDataViewListCtrl`, not the native which is used by default.
So to reproduce this issue you would have to build wxWidgets using the `configure` option `--disable-nativedvc`.

Here are the full stack traces from the assert dialogs.

Assert failure when starting the sample:
```
ASSERT INFO:
../src/gtk/dataobj.cpp(102): assert ""Assert failure"" failed in SetType(): invalid dataformat

BACKTRACE:
[1] wxDataFormat::SetType(wxDataFormatId) .../src/gtk/dataobj.cpp:102
[2] wxDataFormat::wxDataFormat(wxDataFormatId) .../src/gtk/dataobj.cpp:59
[3] wxDataViewMainWindow::EnableDragSource(wxDataFormat const&) .../src/generic/datavgen.cpp:2142
[4] wxDataViewCtrl::EnableDragSource(wxDataFormat const&) .../src/generic/datavgen.cpp:5840
[5] MyFrame::BuildDataViewCtrl(wxPanel*, unsigned int, unsigned long, int) .../samples/dataview/dataview.cpp:824
[6] MyFrame::MyFrame(wxFrame*, wxString const&, int, int, int, int) .../samples/dataview/dataview.cpp:649
[7] MyApp::OnInit() .../samples/dataview/dataview.cpp:386
[8] wxAppConsoleBase::CallOnInit() .../include/wx/app.h:93
[9] wxEntry(int&, wchar_t**) .../src/common/init.cpp:488
[10] wxEntry(int&, char**) .../src/common/init.cpp:517
[11] main .../samples/dataview/dataview.cpp:378
[12] __libc_start_main
[13] _start /home/abuild/rpmbuild/BUILD/glibc-2.22/csu/../sysdeps/x86_64/start.S:121
```

Assert failure when dragging an item:
```
ASSERT INFO:
../src/gtk/dataobj.cpp(102): assert ""Assert failure"" failed in SetType(): invalid dataformat

BACKTRACE:
[1] wxDataFormat::SetType(wxDataFormatId) .../src/gtk/dataobj.cpp:102
[2] wxDataFormat::wxDataFormat(wxDataFormatId) .../src/gtk/dataobj.cpp:59
[3] wxDataViewDropTarget::OnDragOver(int, int, wxDragResult) .../src/generic/datavgen.cpp:1724
[4] wxDropTargetBase::OnEnter(int, int, wxDragResult) .../include/wx/dnd.h:161
[5] target_drag_motion .../src/gtk/dnd.cpp:224
[6] g_closure_invoke
[7] g_signal_emit_valist
[8] g_signal_emit_by_name
[9] gtk_main_do_event
[10] g_main_context_dispatch
[11] g_main_context_iteration
[12] gtk_main_iteration
[13] wxDropSource::DoDragDrop(int) .../src/gtk/dnd.cpp:879
[14] wxDataViewMainWindow::OnMouse(wxMouseEvent&) .../src/generic/datavgen.cpp:5057
[15] wxAppConsoleBase::HandleEvent(wxEvtHandler*, void (wxEvtHandler::*)(wxEvent&), wxEvent&) const .../src/common/appbase.cpp:655
[16] wxAppConsoleBase::CallEventHandler(wxEvtHandler*, wxEventFunctor&, wxEvent&) const .../src/common/appbase.cpp:666
[17] wxEvtHandler::ProcessEventIfMatchesId(wxEventTableEntryBase const&, wxEvtHandler*, wxEvent&) .../src/common/event.cpp:1431
[18] wxEventHashTable::HandleEvent(wxEvent&, wxEvtHandler*) .../src/common/event.cpp:1039
[19] wxEvtHandler::TryHereOnly(wxEvent&) .../src/common/event.cpp:1628
[20] wxEvtHandler::TryBeforeAndHere(wxEvent&) .../include/wx/event.h:4007
[21] wxEvtHandler::ProcessEventLocally(wxEvent&) .../src/common/event.cpp:1561
[22] wxEvtHandler::ProcessEvent(wxEvent&) .../src/common/event.cpp:1534
[23] wxEvtHandler::SafelyProcessEvent(wxEvent&) .../src/common/event.cpp:1650
[24] wxWindowBase::HandleWindowEvent(wxEvent&) const .../src/common/wincmn.cpp:1554
[25] wxWindow::GTKProcessEvent(wxEvent&) const .../src/gtk/window.cpp:1548
[26] gtk_window_motion_notify_callback .../src/gtk/window.cpp:1891
[27] g_closure_invoke
[28] g_signal_emit_valist
[29] g_signal_emit
[30] gtk_propagate_event
[31] gtk_main_do_event
[32] g_main_context_dispatch
[33] g_main_loop_run
[34] gtk_main
[35] wxGUIEventLoop::DoRun() .../src/gtk/evtloop.cpp:59
[36] wxEventLoopBase::Run() .../src/common/evtloopcmn.cpp:87
[37] wxAppConsoleBase::MainLoop() .../src/common/appbase.cpp:377
[38] wxAppConsoleBase::OnRun() .../src/common/appbase.cpp:299
[39] wxAppBase::OnRun() .../src/common/appcmn.cpp:342
[40] wxEntry(int&, wchar_t**) .../src/common/init.cpp:504
[41] wxEntry(int&, char**) .../src/common/init.cpp:517
[42] main .../samples/dataview/dataview.cpp:378
[43] __libc_start_main
[44] _start /home/abuild/rpmbuild/BUILD/glibc-2.22/csu/../sysdeps/x86_64/start.S:121
```

**Comment by vadz** [comment]
Thanks, will be fixed by #22220 when it's merged.
